src/common.h

```diff
@@ -23,8 +23,13 @@
 #ifndef _COMMON_H
 #define _COMMON_H
 
+#include <limits.h>
 #include <stdint.h>
 
+#ifndef OFF_MAX
+	#define OFF_MAX ((1ULL << (sizeof(off_t) * CHAR_BIT - 1)) - 1)
```

Comment:
You need to `#include <sys/types.h>` as `off_t` is defined in this file.

Also I think that result of above `((1ULL ...` should be casted to `off_t` as now `1ULL` results in `unsigned` type (`off_t` is signed) which may produce false-positive warnings when comparing with some other variable of `off_t` type.

Reply:
I would conclude that this should be properly prepared and not be part of this merge then.

I'll keep this merge request as it is, people are free to cherry pick what's wanted and leave out whatever is still worth to be discussed.

Reply:
It is just a simple fix. If you agree I could try to use github feature to edit file in web browser.

Reply:
Done.